src/components/PostDetails.tsx

```diff
@@ -4,104 +1,75 @@
-            repudiandae est voluptatem vel debitis et magnam
-          </p>
-        </div>
-
-        <div className="block">
-          <Loader />
+import { Loader, CommentItem, NewCommentForm } from './index';
+import { Post } from '../types';
+import { useComments } from '../hooks/useComments';
 
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
+type Props = {
+  selectedPost: Post;
+};
 
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
+export const PostDetails: React.FC<Props> = ({ selectedPost }) => {
+  const { id, title, body } = selectedPost;
 
-          <p className="title is-4">Comments:</p>
+  const [isFormOpened, setIsFormOpened] = useState(false);
+  const { comments, errorMessage, isLoading, deleteComment, addComment } =
+    useComments(id);
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
+  useEffect(() => setIsFormOpened(false), [selectedPost]);
 
-            <div className="message-body" data-cy="CommentBody">
-              Some comment
-            </div>
-          </article>
+  return (
+    <div className="content" data-cy="PostDetails">
+      <div className="block">
+        <h2 data-cy="PostTitle">{`#${id}: ${title}`}</h2>
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
+        <p data-cy="PostBody">{body}</p>
+      </div>
 
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
-            <div className="message-body" data-cy="CommentBody">
-              One more comment
-            </div>
-          </article>
+      <div className="block">
+        {isLoading && <Loader />}
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
+        {!isLoading && errorMessage && (
+          <div className="notification is-danger" data-cy="CommentsError">
+            {errorMessage}
+          </div>
+        )}
```

Comment:
Consider adding a retry mechanism or additional user feedback when an error occurs while loading comments. This can improve user experience by providing options to recover from errors.